src/EnergyPlus/api/api.py

```diff
@@ -67,14 +68,10 @@ def __init__(self, running_as_python_plugin: bool = False):
                                          as this is already instantiated for Plugin workflows.
         """
         self.api = cdll.LoadLibrary(api_path())
-        self.api.apiVersionFromEPlus.argtypes = []
+        self.api.apiVersionFromEPlus.argtypes = [c_void_p]
         self.api.apiVersionFromEPlus.restype = c_char_p
-        api_version_from_ep = float(self.api.apiVersionFromEPlus())
-        api_version_defined_here = float(self.api_version())
-        if api_version_defined_here != api_version_from_ep:
-            raise Exception("API version does not match, this API version: %s; E+ is expecting version: %s" % (
-                api_version_defined_here, api_version_from_ep
-            ))
+        # self.state provides access to the main EnergyPlus state management class, instantiated and ready to go
+        self.state = State(self.api)
```

Comment:
Deferred calling the API version check because it needs `state`, and we can't get a `state` until after the API is created...